ee/experimental/graph_utils.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
```

Comment:
general thought: should we maybe have some class for ai/ml based logics?

Reply:
Not sure if it makes sense to merge all the different functions and entities into one class @talboren @Matvey-Kuk